docs: Remove download for settings_reset and note local build (#2543)
zmkfirmware/zmk: docs/docs/troubleshooting/connection-issues.mdx

:::

> ### Building a Reset UF2

Reviewer: I've never been a big fan of the UF2 being in this header. "Building Reset Firmware" maybe?

> ### Building a Reset UF2

Author: ```suggestion
### Building a Reset Firmware
```

> If you are using GitHub Actions to build your firmware as described in the [user setup](../user-setup.mdx), find the `build.yaml` file in your `zmk-config` folder and add an additional settings reset build for the board used by your split keyboard. For example assuming that the config repo is setup for nice!nano v2 with Corne, append the `settings_reset` shield to the `build.yaml` file as follows:

Reviewer: ```suggestion
If you are using GitHub Actions to build your firmware as described in the [user setup](../user-setup.mdx), find the `build.yaml` file in your `zmk-config` folder and add an additional settings reset build for the board used by your split keyboard. For example, assuming that the config repo is setup for nice!nano v2 with Corne, append the `settings_reset` shield to the `build.yaml` file as follows:
```

    shield: settings_reset
```

> If you are building for an onboard controller keyboard that doesn't use a shield, you can add a new `shield:` field with its `board:` entry.

Reviewer: ```suggestion
If you are building for an onboard controller keyboard that doesn't use a shield, add the `shield: settings_reset` field to its `board:` entry.
```

Author: ```suggestion
If you are building for an onboard controller keyboard that doesn't use a shield, add a new `shield:` field to its `board:` entry.
```
I think it's better to keep "a new" rather than "the" to signify the previous non-existence of the field?

> Save the file, commit the changes and push them to GitHub. Download the new firmware zip file build by the latest GitHub Actions job. In it you will find an additional `settings_reset` UF2 image file.

Reviewer: ```suggestion
Save the file, commit the changes and push them to GitHub. Download the new `firmware.zip` file built by the latest GitHub Actions job. In it you will find an additional `settings_reset` UF2 image file.
```

Author: I think the zip file won't be literally called `firmware.zip`, hence the generic "the new firmware zip file" phrasing here. We should do "file that is built" rather than "file build".

> If you use a [local development environment](../development/local-toolchain/setup/index.md) to build firmware instead of GitHub Actions, you can replace your `-DSHIELD=` argument value with `settings_reset` in the [build command](../development/local-toolchain/build-flash.mdx), or add one if the keyboard doesn't normally use one.

Reviewer: ```suggestion
If you use a [local development environment](../development/local-toolchain/setup/index.md) to build firmware instead of GitHub Actions, pass the `-DSHIELD=settings_reset` argument when [building](../development/local-toolchain/build-flash.mdx), omitting all other `-DSHIELD` arguments.
```
